packages/hash/api/src/db/postgres/client.ts

```diff
@@ -87,6 +87,12 @@ export class PostgresClient implements DBClient {
     this.conn = conn;
   }
 
+  jsonSchemaCompiler(conn: Connection) {
+    return new JsonSchemaCompiler(async (url) => {
+      return getJsonSchemaBySchema$id(conn, url);
+    });
+  }
```

Comment:
I appreciate this wasn't introduced as part of this PR, but ideally calls to `jsonSchemaCompiler` should not be made in the implementation of the `PostgresClient`. I don't think schema validation should occur at the database level, but perhaps I'm missing something (cc @CiaranMn).

I realise one of the reasons this was done was likely so that the `jsonSchemaCompiler ` can use the  transaction created in the `createEntityType` db adapter method, so it might be worth considering approaches to moving the transaction creation to a higher level (perhaps into a `createEntityType` method of the `EntityType` model class?). The exact solution likely depends on the decided approach taken [as part of task](https://app.asana.com/0/1201095311341924/1201817757879404/f).

If we agree that we should remove references to the `JsonSchemaCompiler` from the db level, it's probably best to address this in a follow-up PR (not as part of this one).

Reply:
The main reason for it being here is for resolving entityTypes by schema$id directly from the database instead of going though HTTP requests. I agree that there is a certain level of dependency which could be untangled. And you're right it's as it is right now such that it can be a part of the transaction created in `createEntityType` which can hopefully be removed after removing nested transactions.

I will create a follow-up task for simplifying this, blocked by the above linked task.

Reply:
[Asana task](https://app.asana.com/0/1200211978612931/1201849001249402/f) (internal)

Reply:
Thanks, we've discussed this internally and I'm happy for this to be addressed separately once we have refined how we create transactions.